File: web/scripts/audit-pages.py
#!/usr/bin/env python3
"""
Auditoría de páginas de tareas - Captura screenshots y DOM de cada página
"""

from playwright.sync_api import sync_playwright
import json
import os

PAGES = [
    {"name": "MyDay", "app_url": "http://localhost:5177/tareas", "design_url": "http://localhost:3100/sections/task-management/screen-designs/MyDay"},
    {"name": "ProjectList", "app_url": "http://localhost:5177/tareas/proyectos", "design_url": "http://localhost:3100/sections/task-management/screen-designs/ProjectList"},
    {"name": "Kanban", "app_url": "http://localhost:5177/tareas/kanban", "design_url": "http://localhost:3100/sections/task-management/screen-designs/KanbanBoard"},
    {"name": "Dashboard", "app_url": "http://localhost:5177/tareas/dashboard", "design_url": "http://localhost:3100/sections/task-management/screen-designs/KPIDashboard"},
]

OUTPUT_DIR = "/tmp/workhub-audit"

def audit_page(page, name, url, prefix):
    """Captura screenshot y elementos clave de una página"""
    print(f"\n{'='*60}")
    print(f"Auditando: {name} ({prefix})")
    print(f"URL: {url}")
    print(f"{'='*60}")

    page.goto(url)
    page.wait_for_load_state("networkidle")
    page.wait_for_timeout(1000)  # Extra wait for React render

    # Screenshot
    screenshot_path = f"{OUTPUT_DIR}/{name}_{prefix}.png"
    page.screenshot(path=screenshot_path, full_page=True)
    print(f"Screenshot: {screenshot_path}")

    # Capturar elementos clave
    elements = {
        "buttons": [],
        "cards": [],
        "inputs": [],
        "headings": [],
        "stats": [],
    }

    # Botones
    for btn in page.locator("button").all():
        try:
            text = btn.inner_text()[:50] if btn.is_visible() else ""
            if text:
                elements["buttons"].append(text)
        except:
            pass

    # Cards (MUI Paper o divs con shadow)
    card_count = page.locator("[class*='MuiPaper'], [class*='MuiCard']").count()
    elements["cards"].append(f"{card_count} cards detectadas")

    # Inputs
    for inp in page.locator("input, textarea").all():
        try:
            placeholder = inp.get_attribute("placeholder") or ""
            if placeholder:
                elements["inputs"].append(placeholder[:30])
        except:
            pass

    # Headings
    for h in page.locator("h1, h2, h3, h4").all():
        try:
            text = h.inner_text()[:50] if h.is_visible() else ""
            if text:
                elements["headings"].append(text)
        except:
            pass

    # Stats/KPIs (números grandes)
    for stat in page.locator("[class*='stat'], [class*='kpi'], [class*='metric']").all():
        try:
            text = stat.inner_text()[:30] if stat.is_visible() else ""
            if text:
                elements["stats"].append(text)
        except:
            pass

    print(f"\nElementos detectados:")
    print(f"  Botones: {elements['buttons'][:5]}")
    print(f"  Cards: {elements['cards']}")
    print(f"  Inputs: {elements['inputs'][:3]}")
    print(f"  Headings: {elements['headings'][:5]}")

    return elements

def main():
    os.makedirs(OUTPUT_DIR, exist_ok=True)

    results = {}

    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        context = browser.new_context(viewport={"width": 1400, "height": 900})
        page = context.new_page()

        for pg in PAGES:
            name = pg["name"]
            results[name] = {
                "app": audit_page(page, name, pg["app_url"], "app"),
                "design": audit_page(page, name, pg["design_url"], "design"),
            }

        browser.close()

    # Guardar resultados
    with open(f"{OUTPUT_DIR}/audit-results.json", "w") as f:
        json.dump(results, f, indent=2)

    print(f"\n\n{'='*60}")
    print("AUDITORÍA COMPLETADA")
    print(f"Resultados en: {OUTPUT_DIR}")
    print(f"{'='*60}")

if __name__ == "__main__":
    main()

File: web/scripts/audit-pages.ts
import { chromium } from 'playwright'
import * as fs from 'fs'

const PAGES = [
  { name: 'MyDay', appUrl: 'http://localhost:5177/tareas', designUrl: 'http://localhost:3100/sections/task-management/screen-designs/MyDay' },
  { name: 'ProjectList', appUrl: 'http://localhost:5177/tareas/proyectos', designUrl: 'http://localhost:3100/sections/task-management/screen-designs/ProjectList' },
  { name: 'Kanban', appUrl: 'http://localhost:5177/tareas/kanban', designUrl: 'http://localhost:3100/sections/task-management/screen-designs/KanbanBoard' },
  { name: 'Dashboard', appUrl: 'http://localhost:5177/tareas/dashboard', designUrl: 'http://localhost:3100/sections/task-management/screen-designs/KPIDashboard' },
]

const OUTPUT_DIR = '/tmp/workhub-audit'

async function auditPage(page: any, name: string, url: string, prefix: string) {
  console.log(`\n${'='.repeat(60)}`)
  console.log(`Auditando: ${name} (${prefix})`)
  console.log(`URL: ${url}`)
  console.log(`${'='.repeat(60)}`)

  await page.goto(url)
  await page.waitForLoadState('networkidle')
  await page.waitForTimeout(1500) // Extra wait for React render

  // Screenshot
  const screenshotPath = `${OUTPUT_DIR}/${name}_${prefix}.png`
  await page.screenshot({ path: screenshotPath, fullPage: true })
  console.log(`Screenshot: ${screenshotPath}`)

  // Capturar elementos
  const buttons = await page.locator('button').allInnerTexts()
  const headings = await page.locator('h1, h2, h3, h4, h5, h6').allInnerTexts()
  const cards = await page.locator('[class*="MuiPaper"], [class*="MuiCard"], [class*="card"]').count()
  const inputs = await page.locator('input, textarea').count()

  // Chips/badges
  const chips = await page.locator('[class*="MuiChip"], [class*="badge"]').allInnerTexts()

  console.log(`\nElementos detectados:`)
  console.log(`  Botones (${buttons.length}): ${buttons.slice(0, 5).join(', ')}`)
  console.log(`  Headings (${headings.length}): ${headings.slice(0, 5).join(', ')}`)
  console.log(`  Cards: ${cards}`)
  console.log(`  Inputs: ${inputs}`)
  console.log(`  Chips: ${chips.slice(0, 5).join(', ')}`)

  return { buttons, headings, cards, inputs, chips }
}

async function main() {
  if (!fs.existsSync(OUTPUT_DIR)) {
    fs.mkdirSync(OUTPUT_DIR, { recursive: true })
  }

  const results: Record<string, any> = {}

  const browser = await chromium.launch({ headless: true })
  const context = await browser.newContext({ viewport: { width: 1400, height: 900 } })
  const page = await context.newPage()

  for (const pg of PAGES) {
    results[pg.name] = {
      app: await auditPage(page, pg.name, pg.appUrl, 'app'),
      design: await auditPage(page, pg.name, pg.designUrl, 'design'),
    }
  }

  await browser.close()

  // Guardar resultados
  fs.writeFileSync(`${OUTPUT_DIR}/audit-results.json`, JSON.stringify(results, null, 2))

  console.log(`\n\n${'='.repeat(60)}`)
  console.log('AUDITORÍA COMPLETADA')
  console.log(`Resultados en: ${OUTPUT_DIR}`)
  console.log(`${'='.repeat(60)}`)
}

main().catch(console.error)
